### ISSUE TITLE
git submodules

### ISSUE THREAD

**Issue by jneuendorf** [atom-community/autocomplete-paths#148]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When using [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) the contained files are not excluded from the index resulting in an error notification `Maximum file limit of 2000 files exceeded`.

It would be nice to have a possibility to ignore submodules (maybe by introducing a new setting) without having to ignore them explicitly in the `.gitignore` file.

### Steps to Reproduce

1. Add submodule to the project.
2. (Reload the window if notification does not appear.)

**Expected behavior:** Rebuilding the paths cache should work after some kind of configuration for ignoring submodules (other than `.gitignore`).

**Actual behavior:** `Maximum file limit of 2000 files exceeded` shows up. Thus path completion doesn't work.

**Reproduces how often:** 100%

#### Atom

Atom: 1.19.4
Electron: 1.6.9
Chrome: 56.0.2924.87
Node: 7.4.0

#### apm

apm  1.18.4
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.14.1

#### OS

macOS 10.12.6 (16G29)

**Comment by saschagehlich** [comment]
Not sure if this is possible, but I'll look into it

**Comment by jneuendorf** [comment]
Thanks for investigating ;)
In the `tree-view` the span containing the name of the submodule has the class `icon-file-submodule`.
So I'd guess there is a way to find out whether a directory is a submodule or not (using the Atom API).
It looks like [GitRepository::isSubmodule](https://atom.io/docs/api/v1.19.5/GitRepository#instance-isSubmodule) is the way to go:

```javascript
let ignored = false
this._repositories.forEach(repository => {
    if (ignored) return
    if (repository.isPathIgnored(path)) {
        ignored = true
    }
    else if (/*according setting 'ignore submodule' is true AND*/ repository.isSubmodule(path)) {
        ignore = true
    }
})
```

(adjusted from https://github.com/atom-community/autocomplete-paths/blob/f7942b97fe2729a979a60a5bab807361d9117e13/lib/paths-cache.js#L27-L33)

**Comment by saschagehlich** [comment]
Thanks. Landed in 2.9.0
